### ISSUE TITLE
Make it possible to pass a logits processor to `Generator`

### ISSUE THREAD

**Issue by rlouf** [dottxt-ai/outlines#1487]
See for instance https://github.com/dottxt-ai/outlines/pull/1408, but also for internal testing. This is an important design decisions, we need to consider pros & cons of each possible choice.

**Comment by rlouf** [comment]
We could have users pass the logits processor as a keyword-only argument to the `Generator` object. I'm not a fan of introducing a new object or forcing users to know about the internals.

As for the implementation, I like the classmethod solution.
